### PR TITLE
TS HW1: work with types, interfaces

### PR DESCRIPTION
**Pull request by dubinchuk**
HW_11 (TS types, interfaces) for review

### CODE REVIEW

File: TypeScript/HW_11/task 11.2.1.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    projectNames: ['Smoke', 'Regression', 'Re-test'],
+}
+
+function isItEmployee(employee: IEmployee | ItEmployee): employee is ItEmployee {
+    return 'grade' in employee;
+}
+
+function getEmployeeInfo(employee: IEmployee | ItEmployee) {
+    if (isItEmployee(employee)) {
+        if ('address' in employee) {
```
Comment (anatoly-karpovich) [inline TypeScript/HW_11/task 11.2.1.ts:76]: address is not an optional field, so no need to have this check

File: TypeScript/HW_11/task 11.3.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+const item = { id: 5, name: "PC", price: 1500, quantity: 1 };
+
+const cart: IShoppingCart = {
+    items: [
+        { id: 1, name: "Laptop", price: 1000, quantity: 1 },
+        { id: 2, name: "Phone", price: 500, quantity: 2 },
+    ],
+    addItem: (item) => cart.items.push(item),
```
Comment (anatoly-karpovich) [inline TypeScript/HW_11/task 11.3.ts:59]:
```suggestion
    addItem: (item: Item) => cart.items.push(item),
```

File: TypeScript/HW_11/task 11.3.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    removeItem: (id) => {
```
Comment (anatoly-karpovich) [inline TypeScript/HW_11/task 11.3.ts:60]:
```suggestion
    removeItem: (id: number) => {
```

**Review by anatoly-karpovich — APPROVED** [review]

File: TypeScript/HW_11/task 11.2.3.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+//     filter(numbers, (n) => n > 3); // [4, 133]
+//     filter(numbers, (n) => n % 2 == 0); // [2, 4]
+//     Параметры функции: Массив чисел и Анонимная функция, принимающая на вход число и возвращающая логическое значение.
+
+type ArrNum = number[];
+
+const numbers: ArrNum = [1, -5, 2, 3, 4, 133];
+
+function filter(array: ArrNum, callback: (n: number) => boolean) {
+    return array.filter(callback);
```
Comment (VladimirLQA) [inline TypeScript/HW_11/task 11.2.3.ts:15]: Try not to use array method
Implementation with `for` loop looks not bad

**Review by VladimirLQA — APPROVED** [review]

File: TypeScript/HW_11/task 11.2.1.ts (hunk 1 above)
Comment (dubinchuk) [reply TypeScript/HW_11/task 11.2.1.ts]: Fixed

File: TypeScript/HW_11/task 11.3.ts (hunk 3 above)
Comment (dubinchuk) [reply TypeScript/HW_11/task 11.3.ts]: Fixed

File: TypeScript/HW_11/task 11.2.3.ts (hunk 4 above)
Comment (dubinchuk) [reply TypeScript/HW_11/task 11.2.3.ts]: Thanks for clarification. Fixed

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**